rsc/src/main/java/com/cloudera/livy/rsc/rpc/Rpc.java

```diff
@@ -275,10 +273,14 @@ public void operationComplete(ChannelFuture cf) {
       };
 
       dispatcher.registerRpc(id, promise, msg.getClass().getName());
-      synchronized (channelLock) {
```

Comment:
Behavior of `Channel.write()` is different when it's called inside or outside of the event loop. If it's called in the event loop, it writes to the socket immediately. When it's called outside, it enqueues the write to the send buffer. Event loop will write it to the socket later, 1 message at a time.

This lock doesn't prevent the race condition because `Rpc.call()` and `RpcDispatcher.channelRead0()`.